Unable to pass `children` into `<PortalProvider>` or `<Portal>`

Here's a demo of not being able to pass children into `<PortalProvider>`:
```rust
use leptos::*;
use leptos_declarative::prelude::*;

pub fn main() {
    mount_to_body(|cx| {
        view! { cx,
            <Providers>
                <div>"My app."</div>
            </Providers>
        }
    })
}

#[component]
pub fn Providers(cx: Scope, children: Children) -> impl IntoView {
    view! { cx, <PortalProvider>{children(cx)}</PortalProvider> }
}
```
The above code raises the following error:
```
 1  error[E0507]: cannot move out of `children`, a captured variable in an `Fn` closure
   --> src/main.rs:16:5
    |
 15 | pub fn Providers(cx: Scope, children: Children) -> impl IntoView {
    |                             -------- captured outer variable
 16 |     view! { cx, <PortalProvider>{children(cx)}</PortalProvider> }
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^--------^^^^^^^^^^^^^^^^^^^^^^^^
    |     |                            |
    |     |                            variable moved due to use in closure
    |     |                            move occurs because `children` has type `Box<dyn FnOnce(leptos::Scope) -> Fragment>`, which does not implement the `Copy` trait
    |     captured by this `Fn` closure
    |     `children` is moved here
    |
```

And here's an example of not being able to pass children into `<Portal>`:
```rust
use leptos::*;
use leptos_declarative::prelude::*;

struct MyPortalId;

pub fn main() {
    mount_to_body(|cx| {
        view! { cx,
            <PortalProvider>
                <PortalOutput id=MyPortalId/>
                <MyPortal>
                    <p>"I went through the portal!"</p>
                </MyPortal>
            </PortalProvider>
        };
    })
}

#[component]
pub fn MyPortal(cx: Scope, children: Children) -> impl IntoView {
    view! { cx,
        <Portal id=MyPortalId>
            {children(cx)}
        </Portal>
    }
}

```
The above code raises the following error:
```
 1  error[E0507]: cannot move out of `children`, a captured variable in an `Fn` closure
   --> src/main.rs:21:5
    |
 20 |   pub fn MyPortal(cx: Scope, children: Children) -> impl IntoView {
    |                              -------- captured outer variable
 21 | /     view! { cx,
 22 | |         <Portal id=MyPortalId>
 23 | |             {children(cx)}
    | |              --------
    | |              |
    | |              variable moved due to use in closure
    | |              move occurs because `children` has type `Box<dyn FnOnce(leptos::Scope) -> Fragment>`, which does not implement the `Copy` trait
 24 | |         </Portal>
 25 | |     }
    | |     ^
    | |     |
    | |_____captured by this `Fn` closure
    |       `children` is moved here
    |
```
I think these errors might be related to the fact that `portal.rs` uses:
```
type Children = Box<dyn Fn(Scope) -> Fragment>;
```
Whereas `leptos::Children` is of type `Box<dyn FnOnce(Scope) -> Fragment>` (which uses`FnOnce` instead of `Fn`), however, I'm new to Rust, so I don't fully understand these error messages yet.

--- a/src/portal.rs
+++ b/src/portal.rs
@@ -40,8 +40,6 @@ api_planning! {
   }
 }
 
-type Children = Box<dyn Fn(Scope) -> Fragment>;
-
 const CONTEXT_NOT_FOUND_ERROR_MESSAGE: &str =
   "failed to find `PortalCtx`, make sure you are using `<PortalProvider />` \
    somewhere near the root of the app";

